src/com/projectkorra/projectkorra/GeneralMethods.java

```diff
@@ -1719,18 +1721,31 @@ public static boolean isRegionProtectedFromBuildPostCache(final Player player, f
 			}
 
 			if (kingdoms != null && respectKingdoms) {
-				final KingdomPlayer kPlayer = GameManagement.getPlayerManager().getOfflineKingdomPlayer(player).getKingdomPlayer();
-				if (kPlayer.getKingdom() != null) {
-					final SimpleChunkLocation chunkLocation = new SimpleChunkLocation(location.getChunk());
-					final Land land = GameManagement.getLandManager().getOrLoadLand(chunkLocation);
-					final UUID owner = land.getOwnerUUID();
-					if (owner != null) {
-						if (!kPlayer.getKingdom().getKing().equals(owner)) {
-							return true;
+				final KingdomPlayer kPlayer = KingdomPlayer.getKingdomPlayer(player);
+				final Land land = Land.getLand(location);
+				final boolean protectDuringInvasions = ConfigManager.getConfig().getBoolean("Properties.RegionProtection.Kingdoms.ProtectDuringInvasions");
+				if (land != null) {
+					final Kingdom kingdom = land.getKingdom();
+					if (kPlayer.isAdmin()) {
+						return false;
+					}
+					if (land.getInvasion() != null && !protectDuringInvasions) {
+						final Invasion invasion = land.getInvasion();
+						if (invasion.getInvader().equals(kPlayer) && invasion.getDefender().equals(land)) {
+							return false;
 						}
 					}
+					if (!kPlayer.hasKingdom()) {
+						// Player has no kingdom, deny

```

Comment:
We want to allow players who don't have kingdoms but still have access to build in a specific land the ability to build.

Reply:
I don't think that's possible. To allow players to build in a kingdom they have to either (a) be part of that kingdom and have the build permission given to them directly, or (b) be part of a kingdom whose relation permits them to build there. I'm not an expert on Kingdoms but to my knowledge there is no granting build permissions to those who have no kingdom.

Reply:
It actually is. The regulator structure allows players to specify build perms for players that are not in their Kingdom, including players who have no Kingdom.

Reply:
Yes, my bad. I just figured out how to do it.

Reply:
Now adding a player to a Regulator and giving them the build permission will allow them to bend in that chunk.